src/components/date-picker/z-range-picker/index.tsx

```diff
@@ -0,0 +1,595 @@
+
+    prevMonthArrow.setAttribute("aria-label", "Mese precedente");
+    nextMonthArrow.setAttribute("aria-label", "Mese successivo");
+  }
+
+  setDateTimeAriaOptions(calendar, prevMonthArrow, nextMonthArrow) {
+    Array.from(calendar.getElementsByClassName("flatpickr-days")).forEach(
+      (element: HTMLElement) => element.setAttribute("tabindex", "-1")
+    );
+
+    Array.from(calendar.getElementsByClassName("flatpickr-day")).forEach(
+      (element: HTMLElement) => element.setAttribute("role", "button")
+    );
+
+    calendar
+      .getElementsByClassName("flatpickr-monthDropdown-months")[0]
+      .setAttribute("aria-label", "Mese");
+
+    prevMonthArrow.setAttribute("aria-label", "Mese precedente");
+    nextMonthArrow.setAttribute("aria-label", "Mese successivo");
+
+    Array.from(calendar.getElementsByClassName("time24hr")).forEach(
+      (element: HTMLElement) => element.setAttribute("tabindex", "-1")
+    );
+
+    calendar
+      .getElementsByClassName("flatpickr-hour")[0]
+      .setAttribute("aria-label", "Ora");
+    calendar
+      .getElementsByClassName("flatpickr-minute")[0]
+      .setAttribute("aria-label", "Minuti");
+  }
+
+  setFlatpickrPosition() {
+    const toggleHeight = this.element.children[0].clientHeight;
+    this.element.style.setProperty("--toggle-height", `${toggleHeight}px`);
+
+    const flatpickrHeight =
+      this.mode === ZDatePickerMode.dateTime
+        ? ZDatePickerModeValues.DATETIME
+        : ZDatePickerModeValues.DATE;
+
+    const bottom = this.element.getBoundingClientRect().bottom;
+    const overflowBottom = bottom + flatpickrHeight > window.innerHeight;
+    const overflowTop = bottom - flatpickrHeight - toggleHeight < 0;
+
+    if (!overflowTop && overflowBottom) {
+      this.flatpickrPosition = ZDatePickerPosition.top;
+    } else {
+      this.flatpickrPosition = ZDatePickerPosition.bottom;
+    }
+  }
+
+  /** Replace month word to month number */
+  parseDate(date) {
+    const month = date.split(" ")[1];
+    const months = {
+      Gennaio: "01",
+      Febbraio: "02",
+      Marzo: "03",
+      Aprile: "04",
+      Maggio: "05",
+      Giugno: "06",
+      Luglio: "07",
+      Agosto: "08",
+      Settembre: "09",
+      Ottobre: "10",
+      Novembre: "11",
+      Dicembre: "12",
+    };
+
+    const pieces = date.replace(month, months[month]).split(" ");
+    return new Date(
+      parseInt(pieces[2]),
+      parseInt(pieces[1]) - 1,
+      parseInt(pieces[0])
+    );
+  }
+
+  /** Get the current focused input, first or last */
+  getFocusedInput() {
+    if (
+      document.activeElement.matches(
+        `.${RangePickerActiveInput.startInput}.${this.rangePickerId}`
+      )
+    ) {
+      this.activeInput = RangePickerActiveInput.startInput;
+    }
+
+    if (
+      document.activeElement.matches(
+        `.${RangePickerActiveInput.endInput}.${this.rangePickerId}-2`
+      )
+    ) {
+      this.activeInput = RangePickerActiveInput.endInput;
+    }
```

Comment:
modificando come di seguito si potrebbe togliere il metodo `addClassToInput`
```suggestion
    const focusedZInput = document.activeElement.closest('z-input');
    if (!focusedZInput) {
      return;
    }

    if (focusedZInput?.classList.contains(RangePickerActiveInput.startInput)) {
      this.activeInput = RangePickerActiveInput.startInput;
    } else if (focusedZInput.classList.contains(RangePickerActiveInput.endInput)) {
      this.activeInput = RangePickerActiveInput.endInput;
    }
```